reccmp/isledecomp/formats/mz.py

```diff
@@ -0,0 +1,74 @@
+from dataclasses import dataclass
+from pathlib import Path
+import struct
+
+from .image import Image
+
+
+class MZHeaderNotFoundError(ValueError):
+    """MZ magic string not found"""
+
+
+# pylint: disable=too-many-instance-attributes
+@dataclass(frozen=True)
+class ImageDosHeader:
+    # Order is significant!
+    e_magic: int
+    e_cblp: int
+    e_cp: int
+    e_crlc: int
+    e_cparhdr: int
+    e_minalloc: int
+    e_maxalloc: int
+    e_ss: int
+    e_sp: int
+    e_csum: int
+    e_ip: int
+    e_cs: int
+    e_lfarlc: int
+    e_ovno: int
+    e_res: tuple[int, int, int, int]
+    e_oemid: int
+    e_oeminfo: int
+    e_res2: tuple[int, int, int, int, int, int, int, int, int, int]
+    e_lfanew: int
+
+    MAGIC = 0x5A4D  # "MZ"
```

Comment:
In general I think we can just use the constant where it's needed instead of saving it here. If all image files had exactly one magic number/signature then it would make sense, but that's not the case